### DISCUSSION TITLE
Issue with Querying Related Records

### DISCUSSION THREAD

**Discussion by worjs** [prisma/prisma#18359] [Q&A]
Hi everyone!
I have some trouble in selecting some relations,

```prisma
model Recipe {
  id              String    @id @default(uuid())
  userId          String    @map("user_id")
  name            String
  duration        Int       @default(0)

  User User @relation(fields: [userId], references: [id])

  Meal             Meal[]

  @@map("recipe")
}

model Meal {
  id          String   @id @default(uuid())
  userId      String   @map("user_id")
  recipeId    String   @map("recipe_id")
  title       String
  description String?
  photo       String?
  date        DateTime @db.Date

  User   User   @relation(fields: [userId], references: [id])
  Recipe Recipe @relation(fields: [recipeId], references: [id])

  @@index([userId])
  @@map("meal")
}
```

I want to select data like this query
```
SELECT *
FROM Meal
INNER JOIN Recipe ON Recipe.id = Meal.recipeId
WHERE Meal.userId = Recipe.userId
```

i tried it by using meal.findMany method, i think there are no way to do that.

can I get some advice to make this query?

**Comment by ludralph** [comment]
Hi @worjs 👋 

Can you try this query?
```js
const mealsWithRecipes = await prisma.meal.findMany({
  where: {
    userId: {
      equals: recipe.userId,
    },
  },
  include: {
    recipe: {
      select: {
        id: true,
        name: true,
        duration: true,
      },
    },
  },
});
```
If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise, please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Reply by ludralph** [reply]
Hi @worjs 

Do you have any more questions around this issue? If no, it would be really helpful if you could quickly **mark this Discussion as _answered_** to indicate that your issue has been resolved. 

In case you do have any outstanding questions, feel free to let us know and we'd be happy to help!

**Reply by ludralph** [reply]
Hi @worjs, 

Do you have any more questions around this issue? If no, it would be really helpful if you could quickly mark this Discussion as answered to indicate that your issue has been resolved.

In case you do have any outstanding questions, feel free to let us know and we'd be happy to help!
